### DISCUSSION TITLE
make xterm self controlled/controlled

### DISCUSSION THREAD

**Discussion by deflexable** [sourcelair/xterm.js#4972] [Q&A]
how to make xterm self controlled instead of manually writing with `write(string)`. I want it to exhibit the behavior of an `<input />` in HTML

**Comment by jerch** [comment]
You could wrap xterm.js into a web component exposing mandatory settings as attributes, schematically something like this:
```html
<terminal connection="wss://some/websocket/path" cols="120" rows="35" some_other_needed_setting="xyz" />
```
